snapcraft/plugins/catkin.py

```diff
@@ -701,16 +701,15 @@ def _use_in_snap_python(self):
         # Fix all shebangs to use the in-snap python.
         mangling.rewrite_python_shebangs(self.installdir)
 
-        # Also replace the python usage in 10.ros.sh to use the in-snap python.
-        ros10_file = os.path.join(self.rosdir,
-                                  'etc/catkin/profile.d/10.ros.sh')
-        if os.path.isfile(ros10_file):
-            with open(ros10_file, 'r+') as f:
-                pattern = re.compile(r'/usr/bin/python')
-                replaced = pattern.sub(r'python', f.read())
-                f.seek(0)
-                f.truncate()
-                f.write(replaced)
+        # Also replace all the /usr/bin/python calls in etc/catkin/profile.d/
+        # files with the in-snap python
```

Comment:
If I were to nit-pick, it would be more accurate to say the path is stripped so the in-snap Python is used.
Should this be using the absolute path of the snap, though?

Reply:
we cannot know it before hand and it ties the build to the name of the snap which we only do for classic (in very specific cases) as it is unavoidable.

Reply:
if this is not a shebang, it could make use of `SNAP` in the env though

Reply:
I'm not completely clear on whether these are always shell scripts. I suspect they just need to be executable. As a result, I'm more comfortable using `python` from the $PATH (as we've been doing) instead of trying to figure out the language and use environment variables appropriately.